resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -52,6 +52,13 @@ import org.apache.spark.util.{Utils, YarnContainerInfoHelper}
 @ExtendedYarnTest
 class YarnClusterSuite extends BaseYarnClusterSuite {
 
+  private val pythonExecutablePath = {
+    // To make sure to use the same Python executable.
+    val maybePath = TestUtils.getAbsolutePathFromExecutable("python3")
+    assert(maybePath.isDefined)
```

Comment:
maybe we should have a more descriptive error message if this fails so users can easily figure out what's wrong.
It also seems like we should have a better way to specify the python executable then hardcoding to python3 in this test, but that isn't something you changed.

Reply:
If we don't have `python3`, it will not even start to run the tests ... e.g.) see https://github.com/apache/spark/blob/master/dev/run-tests.py#L1 This is rather just a sanity check. But sure, I can add some more messages, though.